src/python/zquantum/core/wip/estimators/estimation_interface.py

```diff
@@ -0,0 +1,44 @@
+from ...circuit import Circuit
+from ...interfaces.backend import QuantumBackend
+from ...measurement import ExpectationValues
+from openfermion import SymbolicOperator
+from typing import List, Callable
+from dataclasses import dataclass
+from typing_extensions import Protocol
+
+
+@dataclass
```

Comment:
Did you consider using an immutable dataclass?
```suggestion
@dataclass(frozen=True)
```